spark3/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -56,48 +83,85 @@ public boolean supportsExternalMetadata() {
   }
 
   @Override
-  public SparkTable getTable(StructType schema, Transform[] partitioning, Map<String, String> options) {
-    // Get Iceberg table from options
-    Configuration conf = SparkSession.active().sessionState().newHadoopConf();
-    Table icebergTable = getTableAndResolveHadoopConfiguration(options, conf);
-
-    // Build Spark table based on Iceberg table, and return it
-    // Eagerly refresh the table before reading to ensure views containing this table show up-to-date data
-    return new SparkTable(icebergTable, schema, true);
+  public Table getTable(StructType schema, Transform[] partitioning, Map<String, String> options) {
```

Comment:
When implementing `SupportsCatalogOptions`, I don't think this will ever be called. Should we remove it and throw `UnsupportedOperationException` instead?

Reply:
`inferPartitioning` method uses it :-( Can we return `Transform[0]` and have Spark do the work?